src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -0,0 +1,248 @@
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.CollectionConfig;
+import com.google.api.codegen.LanguageUtil;
+import com.google.api.codegen.metacode.FieldSetting;
+import com.google.api.codegen.metacode.FieldStructureParser;
+import com.google.api.codegen.metacode.InitCode;
+import com.google.api.codegen.metacode.InitCodeGenerator;
+import com.google.api.codegen.metacode.InitCodeLine;
+import com.google.api.codegen.metacode.InitValueConfig;
+import com.google.api.codegen.metacode.ListInitCodeLine;
+import com.google.api.codegen.metacode.MapInitCodeLine;
+import com.google.api.codegen.metacode.SimpleInitCodeLine;
+import com.google.api.codegen.metacode.StructureInitCodeLine;
+import com.google.api.codegen.viewmodel.FieldSettingView;
+import com.google.api.codegen.viewmodel.FormattedInitValueView;
+import com.google.api.codegen.viewmodel.InitCodeLineView;
+import com.google.api.codegen.viewmodel.InitCodeView;
+import com.google.api.codegen.viewmodel.InitValueView;
+import com.google.api.codegen.viewmodel.ListInitCodeLineView;
+import com.google.api.codegen.viewmodel.MapEntryView;
+import com.google.api.codegen.viewmodel.MapInitCodeLineView;
+import com.google.api.codegen.viewmodel.SimpleInitCodeLineView;
+import com.google.api.codegen.viewmodel.SimpleInitValueView;
+import com.google.api.codegen.viewmodel.StructureInitCodeLineView;
+import com.google.api.codegen.viewmodel.testing.ApiTestAssertView;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class InitCodeTransformer {
+
+  public InitCodeView generateInitCode(MethodTransformerContext context, Iterable<Field> fields) {
+    Map<String, Object> initFieldStructure = createInitFieldStructure(context);
+    InitCodeGenerator generator = new InitCodeGenerator();
+    InitCode initCode =
+        generator.generateRequestFieldInitCode(context.getMethod(), initFieldStructure, fields);
+
+    return InitCodeView.newBuilder()
+        .lines(generateSurfaceInitCodeLines(context, initCode))
+        .fieldSettings(getFieldSettings(context, initCode.getArgFields()))
+        .build();
+  }
+
+  public InitCodeView generateRequestObjectInitCode(MethodTransformerContext context) {
+    Map<String, Object> initFieldStructure = createInitFieldStructure(context);
+    InitCodeGenerator generator = new InitCodeGenerator();
+    InitCode initCode =
+        generator.generateRequestObjectInitCode(context.getMethod(), initFieldStructure);
+
+    return InitCodeView.newBuilder()
+        .lines(generateSurfaceInitCodeLines(context, initCode))
+        .fieldSettings(getFieldSettings(context, initCode.getArgFields()))
+        .build();
+  }
+
+  public List<ApiTestAssertView> generateTestAssertViews(
```

Comment:
By both copying this file from my PR and modifying it at the same time, I can't tell what your modifications are very easily. In future PR updates, please separate the two.